package com.developer.coffeeshopapp.presentation.screens.detailsscreen

import android.R
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.developer.coffeeshopapp.presentation.theme.CharcoalGray
import com.developer.coffeeshopapp.presentation.theme.LightBrown
import com.developer.coffeeshopapp.presentation.theme.LightGray

@Composable
fun SelectSizeChip(sizeText: String, selected: Boolean,onClick:()-> Unit,modifier: Modifier ) {

    Box(
        modifier = modifier
        .background(color = if (selected) LightGray else Color.White, RoundedCornerShape(12.dp))
        .border(1.dp, color =if(selected) LightBrown else LightGray, RoundedCornerShape(12.dp) )
        .height(46.dp)
            .clickable{onClick()},
        contentAlignment = Alignment.Center)
    {
        Text(
            text = sizeText,
            fontSize = 16.sp,
            fontWeight = FontWeight.SemiBold,
            color =if(selected) LightBrown else CharcoalGray

            )
    }
}